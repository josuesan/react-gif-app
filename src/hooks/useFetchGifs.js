import { useState, useEffect, useCallback } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true
  });

  const handleGifs = useCallback(async () => {
    const gifs = await getGifs(category);
    setState({
      data: gifs,
      loading: false
    });
  }, [category]);

  useEffect(() => {
    handleGifs();
  }, [handleGifs]);

  return state;
};
