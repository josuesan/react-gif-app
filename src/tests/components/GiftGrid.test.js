import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('test grif grid', () => {

  const category = 'saitama';
  test('should render success', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when images is loading by useFetchGifs', () => {
    const gifs = [
      {
        id: 'BAC',
        url: 'http:localhost/img.png',
        title: 'any'
      },
      {
        id: 'ABC',
        url: 'http:localhost/img.png',
        title: 'any 2'
      }
    ]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });

});