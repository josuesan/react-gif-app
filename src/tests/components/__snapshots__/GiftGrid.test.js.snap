// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test grif grid should render success 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    saitama
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`test grif grid should show items when images is loading by useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    saitama
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="BAC"
      key="BAC"
      title="any"
      url="http:localhost/img.png"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="any 2"
      url="http:localhost/img.png"
    />
  </div>
</Fragment>
`;
